lib/Controller/RoomController.php

```diff
@@ -2936,13 +2945,50 @@ public function scheduleMeeting(string $calendarUri, int $start, ?array $attende
 			);
 		}
 
+		$ics = $eventBuilder->toIcs();
+		$name = $this->room->getToken() . '-' . $this->timeFactory->getTime() . '.ics';
+		$calendarUri = $this->url->getBaseUrl() . '/remote.php/dav/calendars/' . $user->getUID() . '/' . $calendar->getUri() . '/' . $name;
+
+		try {
+			$loginname = $this->serverSession->get('loginname');
+			$token = $this->random->generate(72, ISecureRandom::CHAR_UPPER . ISecureRandom::CHAR_LOWER . ISecureRandom::CHAR_DIGITS);
+
+			$this->authTokenProvider->generateToken(
+				$token,
+				$user->getUID(),
+				$loginname,
+				null,
+				'Nextcloud Spreed Calendar Integration',
+				IToken::TEMPORARY_TOKEN,
+				IToken::DO_NOT_REMEMBER
+			);
+		} catch (\Exception $e) {
+			$this->logger->debug('Failed to get calendar to schedule a meeting', ['exception' => $e]);
+			return new DataResponse(['error' => 'calendar'], Http::STATUS_BAD_REQUEST);
+		}
+
+		$client = $this->clientService->newClient();
+		$options = [
+			'headers' => [
+				'Content-Type' => 'text/html; charset=UTF-8',
+			],
+			'nextcloud' => [
+				'allow_local_address' => true,
+			],
+			'body' => $ics,
+			'verify' => false,
+			'auth' => [$loginname, $token],
+		];
+
 		try {
-			$eventBuilder->createInCalendar($calendar);
-		} catch (\InvalidArgumentException|CalendarException $e) {
+			$client->put($calendarUri, $options);
+		} catch (\Exception $e) {
 			$this->logger->debug('Failed to get calendar to schedule a meeting', ['exception' => $e]);
 			return new DataResponse(['error' => 'calendar'], Http::STATUS_BAD_REQUEST);
 		}
 
+		$this->authTokenProvider->invalidateToken($token);
```

Comment:
this needs to be in a finally block or also inside the catch?
Should we add a sanity background job to clear those up?